### PR TITLE
feat(gatsby-source-wordpress): use env var to disable catch links

### PR DESCRIPTION
**Pull request by TylerBarnes**
This PR makes the following true:

Because links in html fields are so common in WordPress, `gatsby-source-wordpress` auto-installs `gatsby-plugin-catch-links` for you. In 99% of cases this works well, but for some sites you may need to configure catch-links yourself. You can disable the automatically included version of `gatsby-plugin-catch-links` by setting the environment variable `WORDPRESS_CATCH_LINKS` to a string of `false`. Once it's disabled you can install it yourself and configure it's plugin options in your gatsby-config.js

### CODE REVIEW

File: packages/gatsby-source-wordpress/gatsby-config.js [hunk 1]
```diff
@@ -1,7 +1,10 @@
+const plugins = [`gatsby-plugin-image`,
+`gatsby-plugin-sharp`]
+
+if (process.env.WORDPRESS_CATCH_LINKS !== `false`) {
```
[the remark this answers is not in the document]
Comment (LekoArts) [reply packages/gatsby-source-wordpress/gatsby-config.js:4]: Why not a plugin option?

[the remark this answers is not in the document]
Comment (wardpeet) [reply packages/gatsby-source-wordpress/gatsby-config.js:4]: We should break it in a new major but for now we're stuck with it. Let's do plugin option like @LekoArts mentioned

[the remark this answers is not in the document]
Comment (TylerBarnes) [reply packages/gatsby-source-wordpress/gatsby-config.js:4]: I'm down with a plugin option 👍 the reason I didn't do that was @wardpeet mentioned we should make source-wordpress into a theme, which I think will happen once we turn `gatsby-config` export into a function to conditionally add `catch-links` via a plugin option.
I'm good with that though and then we can remove it in the future

[the remark this answers is not in the document]
Comment (TylerBarnes) [reply packages/gatsby-source-wordpress/gatsby-config.js]: I converted it over to a plugin option if you have time to review again :)

**Review by wardpeet — APPROVED** [review]

**Pull request merged**